src/components/SearchAndFilter.js

```diff
@@ -102,7 +108,19 @@ class SearchAndFilter extends React.Component {
   // Used by the submit button to filter the recipes displayed and closes modal window
   onSubmit = async () => {
     await this.filterRecipes();
-    this.setState({modalOpen: false});
+    this.reset();
+  }
+
+  reset = () => {
```

Comment:
i feel like "Cancel" means a transaction is completely cancelled and you should close the form and return the user to the previous page. If you want to simply clear the form, you can use "Clear" as your button text.

Reply:
"Close" should still close the entire modal/page and return you to the previous page.

Reply:
Added comment for the rest method. Because of how I implemented this entire thing (making the modal depend on the state stored internally), even if I load a new modal, the state is unchanged so I have to reset the state of everything in the modal and then close the form.